recipes/libinterpolate/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class PackageConan(ConanFile):
+    name = "libinterpolate"
+    description = "A C++ interpolation library with a simple interface that supports multiple interpolation methods."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/CD3/libInterpolate"
+    topics = ("math", "spline", "interpolation", "header-only")
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "19.0",
+            "gcc": "7",
+            "clang": "4",
+            "apple-clang": "10",
+        }
+
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("boost/1.80.0", transitive_headers=True)
+        self.requires("eigen/3.3.7", transitive_headers=True)
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Library is only support for Linux. PR's are welcome.")
```

Comment:
Is the libInterpolate only supported for linux, or is this a recipe constraint? I think the call to action to submit a PR to add windows/macos support here is only appropriate if you're suggesting that this is the current state of the recipe, and the recipe can be improved.

If what you mean here is that PRs would be welcome to add windows/macos support to the upstream library, I think this should be more explicit and provide a link to the upstream issue tracker/github repo.

Reply:
The problem is with the upstream library. It fails to compile on Windows and MacOS in the CI. I have not ran into the build errors before, so I will need to dig into it, but I don't have a MacOS machine, so I will only be able to look into the Windows errors.

I will clarify, thank you.